### PR TITLE
Refactor Docker action to Javascript action

### PR DESCRIPTION
**Pull request by martijngastkemper**
I always wondered why this action takes so long to bootstrap. A Docker container must be started and `npm install` every time it runs. So I started to read about Github Actions and found out that there are [three types of actions](https://docs.github.com/en/actions/creating-actions/about-custom-actions#types-of-actions): Docker, Javascript, and Composite. This PR refactors this action from a Docker to Javascript type. The action is now able to run directly on the runner, so no Docker containers are involved.

What did I do:

- Replace `Dockerfile` with `action.yml`.
- To prevent an `npm ci` call every time the action runs I added `@vercel/ncc`.
- It's best practice to add `dist/` to the repository: https://docs.github.com/en/actions/creating-actions/releasing-and-maintaining-actions#about-javascript-actions *
- Added workflow `test.yml` to run the unit tests and run the action on every commit.
- Added workflow `check-dist.yml`: copied from https://github.com/actions/javascript-action

The result: the action finishes in a few seconds instead of 45.

\* I've tried to use [JasonEtco/build-and-tag-action](https://github.com/JasonEtco/build-and-tag-action), which updates the tag with compiled JS files. It would prevent adding `dist/` to the repo. But it fails in combination with `wordpos`: https://github.com/JasonEtco/build-and-tag-action/issues/20

### CODE REVIEW

File: .github/workflows/check-dist.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# When you reference an action with `uses:` in a workflow,
+# `index.js` is the code that will run.
+# For our project, we generate this file through a build process from other source files.
+# We need to make sure the checked-in `index.js` actually matches what we expect it to be.
+name: Check dist/
+
+on:
+  push:
+    branches:
+      - main
```
Comment (harmenjanssen) [inline .github/workflows/check-dist.yml]: This repo doesn't use a `main` branch (yet)! 
Let's fix that outside this PR.

**Review by harmenjanssen — APPROVED** [review]
Wow, that's amazing! 
Thanks a lot. 🙂 

I saw one reference to branch `main`, but this repo uses `master` still. If you change that, I will update the repo (and workflow) to `main` after merging.

File: .github/workflows/check-dist.yml (hunk 1 above)
Comment (martijngastkemper) [reply .github/workflows/check-dist.yml]: Fixed!

**Comment by martijngastkemper** [comment]
Gentle reminder: don't actually remove `master`, without it all current workflows will break...

**Comment by harmenjanssen** [comment]
Thanks, I needed that reminder... 😬 
In that case I'm leaving it aat `master`, otherwise it would just be confusing.

**Pull request merged**